### PR TITLE
Feature/friend list

### PR DESCRIPTION
**Pull request by Tetardfurieux**
Add invitation + Test coverage

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/healthplay/HomeScreenActivityTest.java [hunk 1]
```diff
@@ -14,20 +15,43 @@
...
-         //fragment.check(ViewAssertions.matches(ViewMatchers.withId(R.id.profileSetupActivity)));
+    }
+
+    @Test
+    public void isOnline(){
+        Database database = new Database();
+        database.readField(FirebaseAuth.getInstance().getCurrentUser().getUid(), "onlineStatus", new OnCompleteListener<DataSnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<DataSnapshot> task) {
+                assertThat(task.getResult().getValue(String.class), Matchers.equalTo("online"));
```
Comment (Zephyr75) [inline app/src/androidTest/java/ch/epfl/sdp/healthplay/HomeScreenActivityTest.java:53]: Very nice way of testing online activity

File: app/src/main/java/ch/epfl/sdp/healthplay/chat/AdapterChat.java [hunk 2]
```diff
@@ -124,22 +141,79 @@ public void onBindViewHolder(@NonNull MyHolder holder, @SuppressLint("RecyclerVi
...
+                                }
+                                database.getLobbyPlayerCount(lobbyName, Database.MAX_NBR_PLAYERS, task3 -> {
+                                    if (!task3.isSuccessful()) {
+                                        Log.e("ERROR", "Lobby does not exist!");
+                                    }
+                                    if (Math.toIntExact((long) task2.getResult().getValue()) < Math.toIntExact((long) task3.getResult().getValue())) {
+                                        database.addUserToLobby(lobbyName, username);
+                                        context.startActivity(intent);
+                                    } else {
+                                        Snackbar.make(holder.itemView,"The lobby is full", Snackbar.LENGTH_SHORT ).show();
```
Comment (Zephyr75) [inline app/src/main/java/ch/epfl/sdp/healthplay/chat/AdapterChat.java:171]: Add translation to this snackbar

**Review by Zephyr75 — CHANGES_REQUESTED** [review]
Very nice comments everywhere and interestiong tests! Just a couple of quick comments to look at.

File: app/src/main/java/ch/epfl/sdp/healthplay/chat/AdapterChat.java [hunk 3]
```diff
@@ -43,6 +52,7 @@ public class AdapterChat extends RecyclerView.Adapter<AdapterChat.MyHolder> {
     private String imageUrl;
     private FirebaseUser firebaseUser;
```
Comment (emonnin-epfl) [inline app/src/main/java/ch/epfl/sdp/healthplay/chat/AdapterChat.java:53]: firebaseUser and database should be final.

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java [hunk 4]
```diff
@@ -650,13 +652,14 @@ public void deleteLobby(String name) {
      * Get the friend list of the user
      * @return a map of String to Boolean
      */
-    public Map<String, Boolean> getFriendList() {
-        Map<String, Boolean> outputMap = new HashMap<>();
+    //TODO make it work
+    public Map<String, String> getFriendList() {
+        Map<String, String> outputMap = new HashMap<>();
         readField(FirebaseAuth.getInstance().getCurrentUser().getUid(), "friends", new OnCompleteListener<DataSnapshot>() {
```
Comment (emonnin-epfl) [inline app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java:658]: Isn't there a friend constant in the database class?

**Review by emonnin-epfl — CHANGES_REQUESTED** [review]

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java [hunk 5]
```diff
@@ -763,4 +766,78 @@ private void getLeaderBoard(OnCompleteListener<DataSnapshot> onCompleteListener,
...
+
+        //Create the message with the informations about the sender, receiver ...
+        HashMap<String, Object> hashMap = new HashMap<>();
+        hashMap.put("sender", senderId);
+        hashMap.put("receiver", receiverId);
+        hashMap.put("message", message);
+        hashMap.put("timestamp", timestamp);
+        hashMap.put("type", "invitation");
+        //Add the message to the database
+        mDatabase.child("Chats").push().setValue(hashMap);
```
[the remark this answers is not in the document]
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java:826]: +1

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java (hunk 4 above)
Comment (Tetardfurieux) [reply app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java:658]: Yes indeed

**Review by emonnin-epfl — APPROVED** [review]

**Review by Zephyr75 — APPROVED** [review]

**Pull request merged**